shell/platform/android/io/flutter/plugin/common/MethodChannel.java

```diff
@@ -152,6 +156,54 @@ public void setMethodCallHandler(final MethodCallHandler handler) {
         void notImplemented();
     }
 
+    /**
+     * An exception that {@link MethodCallHandler} implementations may throw in
+     * {@link MethodCallHandler#onMethodCall(MethodCall, Result)} that will be caught by the
+     * channel implementation and sent back to Flutter as a {@code PlatformException}. Metadata
+     * in this exception class will be translated to corresponding fields in the Dart exception.
+     *
+     * <p>Other uncaught exceptions thrown by {@code MethodCallHandler} implementations will also
+     * be sent back to Flutter as a {@code PlatformException}, but they will have a generic
+     * "error" code and no error details.</p>
+     */
+    public static final class ResultException extends RuntimeException {
```

Comment:
Another option would be for the engine to define `Callable` and/or `Runnable` wrappers that provide the same exception handling so that we could get rid of `result.error()`...  In which case, the addition of this exception would be a step in the right direction.